services/ui-src/src/components/tables/getEntityStatus.tsx

```diff
@@ -75,3 +82,78 @@ export const getEntityStatus = (report: ReportShape, entity: EntityShape) => {
     return false;
   }
 };
+
+export const getInitiativeStatus = (
+  report: ReportShape,
+  entity: EntityShape
+) => {
+  // Direct pull of the initiative formTemplate json chunk
+  const reportRoute = report.formTemplate
+    .routes[3] as unknown as ModalOverlayReportPageShape;
+
+  //get the intiative report child
+  const reportChild: EntityDetailsDashboardOverlayShape =
+    reportRoute.children![1];
+
+  if (reportChild.entitySteps) {
+    const entitySteps: (EntityDetailsOverlayShape | OverlayModalPageShape)[] =
+      reportChild.entitySteps;
+    const stepStatuses = entitySteps.map((step) => {
+      return getInitiativeDashboardStatus(step, entity);
+    });
+
+    return stepStatuses.every((field: any) => field);
+  }
+
+  return false;
+};
+
+//NOTE: this function works on the assumption that the fieldData saved is validated
+export const getInitiativeDashboardStatus = (
+  formEntity: any,
```

Comment:
Could the `formEntity` be a type `AnyObject` instead of `any` or is it possible it won't be an object at some point?

Reply:
that's a good point, going to capture this and update it in a cleanup branch. thanks Karla!